File: prueba-frontend/src/app/modules/intranet/services/cliente.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { ClienteBuscar } from '../dto/request/cliente-buscar';
import { Cliente } from '../dto/response/cliente';
import { OutResponse } from '../dto/response/out.response';

@Injectable()
export class ClienteService {

  constructor(private http: HttpClient) { }

  public registrarCliente(req: Cliente): Observable<OutResponse<Cliente>> {
    return this.http.post<OutResponse<Cliente>>(`${environment.pruebaBackendUrl}/cliente/registrarCliente`, req);
  }

  public modificarCliente(req: Cliente): Observable<OutResponse<any>> {
    return this.http.post<OutResponse<any>>(`${environment.pruebaBackendUrl}/cliente/modificarCliente`, req);
  }

  public listarCliente(req: Cliente): Observable<OutResponse<Cliente[]>> {
    return this.http.post<OutResponse<Cliente[]>>(`${environment.pruebaBackendUrl}/cliente/listarCliente`, req);
  }

  public eliminarClienteLogica(req: Cliente): Observable<OutResponse<any>> {
    return this.http.post<OutResponse<any>>(`${environment.pruebaBackendUrl}/cliente/eliminarClienteLogica`, req);
  }

  public listarClientePaginado(req: ClienteBuscar): Observable<OutResponse<Cliente[]>> {
    return this.http.post<OutResponse<Cliente[]>>(`${environment.pruebaBackendUrl}/cliente/listarClientePaginado`, req);
  }
}

File: prueba-backend/src/main/java/com/empresa/proyecto/service/ClienteService.java
package com.empresa.proyecto.service;

import java.util.List;

import com.empresa.proyecto.dto.request.ClienteBuscar;
import com.empresa.proyecto.dto.response.Cliente;
import com.empresa.proyecto.dto.response.OutResponse;

public interface ClienteService {

	public OutResponse<Cliente> registrarCliente(Cliente c);

	public OutResponse<?> modificarCliente(Cliente c);

	public OutResponse<List<Cliente>> listarCliente(Cliente r);

	public OutResponse<?> eliminarClienteLogica(Cliente r);

	public OutResponse<?> eliminarCliente(Cliente c);

	public OutResponse<List<Cliente>> listarClientePaginado(ClienteBuscar r);
}
